src/algebra/ring/equiv.lean

```diff
@@ -295,31 +293,47 @@ This is the `ring_equiv` version of `equiv.Pi_congr_right`, and the dependent ve
 -/
 @[simps apply]
 def Pi_congr_right {ι : Type*} {R S : ι → Type*}
-  [Π i, semiring (R i)] [Π i, semiring (S i)]
+  [Π i, non_unital_non_assoc_semiring (R i)] [Π i, non_unital_non_assoc_semiring (S i)]
   (e : Π i, R i ≃+* S i) : (Π i, R i) ≃+* Π i, S i :=
 { to_fun := λ x j, e j (x j),
   inv_fun := λ x j, (e j).symm (x j),
   .. @mul_equiv.Pi_congr_right ι R S _ _ (λ i, (e i).to_mul_equiv),
   .. @add_equiv.Pi_congr_right ι R S _ _ (λ i, (e i).to_add_equiv) }
 
 @[simp]
-lemma Pi_congr_right_refl {ι : Type*} {R : ι → Type*} [Π i, semiring (R i)] :
+lemma Pi_congr_right_refl {ι : Type*} {R : ι → Type*} [Π i, non_unital_non_assoc_semiring (R i)] :
   Pi_congr_right (λ i, ring_equiv.refl (R i)) = ring_equiv.refl _ := rfl
 
 @[simp]
 lemma Pi_congr_right_symm {ι : Type*} {R S : ι → Type*}
-  [Π i, semiring (R i)] [Π i, semiring (S i)]
+  [Π i, non_unital_non_assoc_semiring (R i)] [Π i, non_unital_non_assoc_semiring (S i)]
   (e : Π i, R i ≃+* S i) : (Pi_congr_right e).symm = (Pi_congr_right $ λ i, (e i).symm) := rfl
 
 @[simp]
 lemma Pi_congr_right_trans {ι : Type*} {R S T : ι → Type*}
-  [Π i, semiring (R i)] [Π i, semiring (S i)] [Π i, semiring (T i)]
+  [Π i, non_unital_non_assoc_semiring (R i)] [Π i, non_unital_non_assoc_semiring (S i)]
+  [Π i, non_unital_non_assoc_semiring (T i)]
   (e : Π i, R i ≃+* S i) (f : Π i, S i ≃+* T i) :
   (Pi_congr_right e).trans (Pi_congr_right f) = (Pi_congr_right $ λ i, (e i).trans (f i)) := rfl
 
+end non_unital_semiring
+
+section semiring
+
+variables [non_assoc_semiring R] [non_assoc_semiring S] (f : R ≃+* S) (x y : R)
+
+/-- A ring isomorphism sends one to one. -/
+protected lemma map_one : f 1 = 1 := map_one f
+
+variable {x}
+
+protected lemma map_eq_one_iff : f x = 1 ↔ x = 1 := mul_equiv_class.map_eq_one_iff f
+
+lemma map_ne_one_iff : f x ≠ 1 ↔ x ≠ 1 := mul_equiv_class.map_ne_one_iff f
+
 end semiring
 
-section
+section ring
 
 variables [non_assoc_ring R] [non_assoc_ring S] (f : R ≃+* S) (x y : R)
```

Comment:
Do the lemmas here generalize to non-unitarity too?

Reply:
I guess technically `ring_equiv.map_neg` does, but we have `map_neg` anyway.